app/src/main/java/com/orange/ods/app/ui/Screen.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *  Copyright 2021 Orange
+ *
+ *  Use of this source code is governed by an MIT-style
+ *  license that can be found in the LICENSE file or at
+ *  https://opensource.org/licenses/MIT.
+ * /
+ */
+
+package com.orange.ods.app.ui
+
+import android.os.Bundle
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.text.input.TextFieldValue
+import com.orange.ods.app.R
+import com.orange.ods.app.ui.components.ComponentsNavigation
+import com.orange.ods.app.ui.components.ComponentsNavigation.ComponentDemoRoute
+import com.orange.ods.app.ui.components.ComponentsNavigation.ComponentDetailRoute
+import com.orange.ods.app.ui.components.ComponentsNavigation.ComponentVariantDemoRoute
+import com.orange.ods.app.ui.components.Variant
+import com.orange.ods.app.ui.guidelines.GuidelinesNavigation
+import com.orange.ods.app.ui.utilities.UiString
+import com.orange.ods.compose.component.content.OdsComponentContent
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
+/**
+ * Returns the [Screen] corresponding to the given [route].
+ */
+fun getScreen(route: String?, args: Bundle?): Screen? = route?.let {
+    val matchElementRouteResult = Regex("^(.+)/\\{.+\\}$").find(route)
+    if (matchElementRouteResult != null) {
+        // Specific element route -> get element id
+        val (routeRoot) = matchElementRouteResult.destructured
+        when (routeRoot) {
+            ComponentDetailRoute, ComponentDemoRoute -> Screen.Component(args?.getLong(ComponentsNavigation.ComponentIdKey))
+            ComponentVariantDemoRoute -> Screen.ComponentVariant(args?.getLong(ComponentsNavigation.ComponentVariantIdKey))
+            else -> null
+        }
+    } else {
+        // Simple route
+        val screens = Screen::class.nestedClasses.mapNotNull { kClass -> kClass.objectInstance as? Screen }
+        screens.firstOrNull { screen -> screen.route == route }
+    }
+}
+
+/**
+ * Defines application common changing elements for each screen.
+ * It allows to manage top app bar display according to the screen.
+ */
+sealed class Screen(
+    val route: String,
+    val isLargeAppBar: Boolean = false,
+    val title: UiString? = null,
+    val type: ScreenType = ScreenType.Default,
```

Comment:
This field is a duplicated information for the search screen:

```kotlin
data object Search : Screen(
    route = MainDestinations.SearchRoute,
    type = ScreenType.Search
)
```

In that case we could simply check the `Screen` class to see that it's the search screen.

Moreover, the `ScreenType` seems to be only used to configure the app bar.

Thus I wonder if we could get rid of it. It could be replaced with a `isHome` property:

```kotlin
val isHome: Boolean
    get() = this in listOf(Guidelines, Components, Modules, About)
```

and a `isCustomAppBar` property for instance.